core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupTemplate.java

```diff
@@ -52,74 +52,118 @@ public void initializeLogging(LogConfig config) {
-                periodicSizeRotatingFileHandler.setSuffix(rotationConfig.fileSuffix.get());
-                periodicSizeRotatingFileHandler.setRotateSize(rotationConfig.maxFileSize.get().asLongValue());
-                periodicSizeRotatingFileHandler.setRotateOnBoot(rotationConfig.rotateOnBoot);
-                periodicSizeRotatingFileHandler.setMaxBackupIndex(rotationConfig.maxBackupIndex);
-                handler = periodicSizeRotatingFileHandler;
-            } else if (rotationConfig.maxFileSize.isPresent()) {
-                SizeRotatingFileHandler sizeRotatingFileHandler = new SizeRotatingFileHandler(
-                        rotationConfig.maxFileSize.get().asLongValue(), rotationConfig.maxBackupIndex);
-                sizeRotatingFileHandler.setRotateOnBoot(rotationConfig.rotateOnBoot);
-                handler = sizeRotatingFileHandler;
-            } else if (rotationConfig.fileSuffix.isPresent()) {
-                PeriodicRotatingFileHandler periodicRotatingFileHandler = new PeriodicRotatingFileHandler();
-                periodicRotatingFileHandler.setSuffix(rotationConfig.fileSuffix.get());
-                handler = periodicRotatingFileHandler;
-            }
+            initFileHandler(config, errorManager, filterElements, handlers);
+        }
+
+        if (config.syslog.enable) {
+            initSyslogHandler(config, errorManager, filterElements, handlers);
+        }
+
+        InitialConfigurator.DELAYED_HANDLER.setHandlers(handlers.toArray(EmbeddedConfigurator.NO_HANDLERS));
+    }
+
+    private ErrorManager initConsoleHandler(LogConfig config, ErrorManager errorManager, List<LogCleanupFilterElement> filterElements, ArrayList<Handler> handlers) {
+        final PatternFormatter formatter;
+        if (config.console.color && System.console() != null) {
+            formatter = new ColorPatternFormatter(config.console.darken, config.console.format);
+        } else {
+            formatter = new PatternFormatter(config.console.format);
+        }
+        final ConsoleHandler handler = new ConsoleHandler(formatter);
+        handler.setLevel(config.console.level);
+        handler.setErrorManager(errorManager);
+        handler.setFilter(new LogCleanupFilter(filterElements));
+        if (config.console.async.enable) {
+            final AsyncHandler asyncHandler = new AsyncHandler(config.console.async.queueLength);
+            asyncHandler.setOverflowAction(config.console.async.overflow);
+            asyncHandler.addHandler(handler);
+            asyncHandler.setLevel(config.console.level);
+            handlers.add(asyncHandler);
+        } else {
+            handlers.add(handler);
+        }
+        errorManager = handler.getLocalErrorManager();
+        return errorManager;
+    }
+
+    private void initFileHandler(LogConfig config, ErrorManager errorManager, List<LogCleanupFilterElement> filterElements, ArrayList<Handler> handlers) {
```

Comment:
Maybe pass `FileConfig` as a parameter?

Reply:
I am thinking of naming the method `configureFileHandler(...)`, what do you think?

Reply:
I'm agree with you, i used IDE method extraction so it's not optimized :
configureFileHandler(...)  yes, it's a better method naming.